### DISCUSSION TITLE
Use of Expat <2.6.0 with `XML_DTD` un(!)defined is vulnerable to CVE-2023-52426 (denial of service)

### DISCUSSION THREAD

**Discussion by hartwork** [osmandapp/OsmAnd#19040] [General]
Hi!

It has come to my attention that Expat is used with `XML_DTD` undefined here…

https://github.com/osmandapp/OsmAnd-core/blob/e88a878300eb80c7a526ae54c245b6e22f1bf8cf/externals/expat/expat_config.h.ios#L66

… which is a security problem with all releases prior to Expat 2.6.0 that was [released today](https://blog.hartwork.org/posts/expat-2-6-0-released/). There are at least two known ways forward for OsmAnd to address this issue: (a) enabling `XML_DTD` or (b) making the build require Expat >=2.6.0. For both then to release new binaries. I know little about OsmAnd so I may be missing something. What do you think?

Best, Sebastian

**Comment by vshcherb** [comment]
Yes thank good point, we need to schedule an update
